docs/contributing/Style-guide-for-Flutter-repo.md

```diff
@@ -5,7 +5,7 @@
 Optimize for readability. Write detailed documentation.
 Make error messages useful.
 Never use timeouts or timers.
-Avoid `is`, `print`, `part of`, `extension` and `_`.
+Avoid hidden dependencies: powerful systems are composed of simple, reusable parts.
```

Comment:
I don't have strong feelings about the line I added here, but I do feel that the previous version is outdated.

<br>

> Avoid `is`

This advice [applies to widgets](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-interleaving-multiple-concepts-together), but IMO it's a blanket statement that ignores the overall advantages of type promotion.

<br>

> Avoid `print`

This is covered by a [linter rule](https://dart.dev/lints/avoid_print).

<br>

> Avoid `part of`

I agree that this is good advice, but I seldom ever see a non-Googler creating a new file; IMO we are better off keeping [the below section](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-library-and-part-of) but removing it from the summary.

<br>

> Avoid `extension`

That's what this PR is about!

<br>

> Avoid `_`

Prior to Dart 3.0, `_` was exclusively used for [anonymous parameter names](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-anonymous-parameter-names), but now it's also meant to designate a [wildcard pattern](https://dart.dev/language/pattern-types#wildcard).

Reply:
This top section is intended to be just a brief tl;dr of whatever the most common mistakes are. For example, there was a time where people were frequently uploading PRs that used `is` instead of carefully designed polymorphism, so the `is` advice got promoted to the top here, and that helped. We should feel free to keep updating this section to reflect whatever is the most useful advice to give at the moment.